src/set.rs

```diff
@@ -0,0 +1,229 @@
+//! A concurrent hash set backed by DashMap.
+
+use std::borrow::Borrow;
+use std::collections::hash_map::RandomState;
+use std::hash::{BuildHasher, Hash};
+
+use crate::DashMap;
+
+/// A concurrent hash set backed by DashMap.
+pub struct DashSet<T, S = RandomState> {
+    map: DashMap<T, (), S>,
+}
+
+impl<'a, T: 'a + Eq + Hash> DashSet<T, RandomState> {
+    /// Creates a new DashSet with a capacity of 0.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use dashmap::set::DashSet;
+    ///
+    /// let games = DashSet::new();
+    /// games.insert("Veloren");
+    /// ```
+    pub fn new() -> Self {
+        DashSet::with_hasher(RandomState::default())
+    }
+
+    /// Creates a new DashSet with a specified starting capacity.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use dashmap::set::DashSet;
+    ///
+    /// let ds = DashSet::with_capacity(2);
+    /// ds.insert(2);
+    /// ds.insert(4);
+    /// ```
+    #[inline]
+    pub fn with_capacity(capacity: usize) -> Self {
+        DashSet::with_capacity_and_hasher(capacity, RandomState::default())
+    }
+}
+
+impl<'a, T: 'a + Eq + Hash, S: BuildHasher + Clone> DashSet<T, S> {
+    /// Creates a new DashSet with a capacity of 0 and the provided hasher.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use dashmap::set::DashSet;
+    /// use std::collections::hash_map::RandomState;
+    ///
+    /// let s = RandomState::new();
+    /// let games = DashSet::with_hasher(s);
+    /// games.insert("Veloren");
+    /// ```
+    pub fn with_hasher(hasher: S) -> Self {
+        Self {
+            map: DashMap::with_hasher(hasher),
+        }
+    }
+
+    /// Creates a new DashSet with a specified starting capacity and hasher.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use dashmap::set::DashSet;
+    /// use std::collections::hash_map::RandomState;
+    ///
+    /// let s = RandomState::new();
+    /// let ds = DashSet::with_capacity_and_hasher(2, s);
+    /// ds.insert(2);
+    /// ds.insert(4);
+    /// ```
+    #[inline]
```

Comment:
`#[inline]` is a copy-paste error, I'm not sure what ones of these functions should be marked inline, if any. I'll remove this or add others at the same time rather than fixing this now.